includes/documents/class-wcpdf-invoice.php

```diff
@@ -103,25 +103,50 @@ public function init_number() {
 			return $invoice_number;
 		}
 
-		$number_store_method = WPO_WCPDF()->settings->get_sequential_number_store_method();
-		$number_store_name = apply_filters( 'wpo_wcpdf_document_sequential_number_store', 'invoice_number', $this );
-		$number_store = new Sequential_Number_Store( $number_store_name, $number_store_method );
-		// reset invoice number yearly
-		if ( isset( $this->settings['reset_number_yearly'] ) ) {
-			$current_year = date("Y");
-			$last_number_year = $number_store->get_last_date('Y');
-			// check if we need to reset
-			if ( $current_year != $last_number_year ) {
-				$number_store->set_next( apply_filters( 'wpo_wcpdf_reset_number_yearly_start', 1, $this ) );
-			}
+		if( ! empty( $document_date = $this->get_date() ) ) {
+			$number_store_method = WPO_WCPDF()->settings->get_sequential_number_store_method();
+			$number_store_name   = $this->store_name( $document_date );
```

Comment:
I think this could be simplified by renaming the method `get_sequential_number_store()` and letting it return the instance. You don't need to pass `$document_date` either since it's an internal property.

Reply:
In the document or in the settings class?

Reply:
in the document class. This:
https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/97ae3403ade900fc26c3d703fa814550d3b6583c/includes/documents/class-wcpdf-invoice.php#L106
Would then also be in that same `get_sequential_number_store()` method, keeping it clean